src/features/QueryStatus.tsx

```diff
@@ -0,0 +1,53 @@
+import { UseQueryResult } from "@tanstack/react-query";
+import { Children, ReactNode } from "preact/compat";
+
+type Props = {
+  children: ReactNode;
+
+  /** Tanstack Query object */
+  query: UseQueryResult<unknown, Error>;
+
+  /** Given query data is considered 'empty' when method is true */
+  isEmpty: (arg: any) => boolean;
+};
+
+export const Loading = ({ children }) => <>{children}</>;
```

Comment:
Seems redundant, but it is necessary for these to be separate named components to decern the type within the array of children.